### DISCUSSION TITLE
Specifying upstreams for a domain

### DISCUSSION THREAD

**Discussion by majkassab** [AdguardTeam/AdGuardHome#4781] [Q&A] [answered]
Hello,

If I write the following syntax in the custom filtering rules can be work?

[/majkassab.com/]1.1.1.1

Because I want to specify a list of domains to use a specific public dns

Thank you

**Comment by EugeneOne1** [comment, marked as answer]
@majkassab, hello.  The mentioned string is the example of domain-specific upstream syntax.  Check the [wiki](https://github.com/AdguardTeam/AdGuardHome/wiki/Configuration#upstreams-for-domains) about that.  The exact string `[/majkassab.com/]1.1.1.1` put into the upstreams along with some common one will make AdGuardHome to resolve the `majkassab.com` domain and all it's subdomains via Cloudflare's DNS (`1.1.1.1`).  Any other domain will again be resolved via the common upstream server.

Also, if you only want the `majkassab.com` to be resolved via `1.1.1.1`, and it's subdomains to be resolved via common one, you should additionally put the following line to the upstreams list:
```
[/*.majkassab.com/]#
```

**Reply by majkassab** [reply]
Thank you so much for your help
